dnf/plugins/leaves/dnf-command-leaves.c

```diff
@@ -332,7 +345,7 @@ dnf_command_leaves_run (DnfCommand      *cmd,
   g_ptr_array_sort (pkgs, gptrarr_dnf_package_cmp);
 
   // build the directed graph of dependencies
-  g_autoptr(GPtrArray) graph = build_graph (query, pkgs);
+  g_autoptr(GPtrArray) graph = build_graph (query, pkgs, opt_recommends);
```

Comment:
Thanks for the changes. In the meantime, I discussed the behaviour of the `leaves` command with more people. They prefer to respect the `install_weak_deps` configuration option instead of introducing another command argument.
If `install_weak_deps` is set to true, they want 'recommends' to be installed. So `leaves` should consider the recommends as required.
Therefore, use `dnf_context_get_install_weak_deps()` instead of the new argument.

`g_autoptr(GPtrArray) graph = build_graph (query, pkgs, dnf_context_get_install_weak_deps ());`

`install_weak_deps` can be set in the configuration file or on the command line using `--setopt=install_weak_deps=1/0`.